markdown/core.py

```diff
@@ -442,18 +445,14 @@ def convertFile(
                 output_file.close()
             else:
                 writer = codecs.getwriter(encoding)
-                output_file = writer(output, errors="xmlcharrefreplace")
-                output_file.write(html)
+                output_writer = writer(output, errors="xmlcharrefreplace")
+                output_writer.write(html)
                 # Don't close here. User may want to write more.
         else:
             # Encode manually and write bytes to stdout.
-            html = html.encode(encoding, "xmlcharrefreplace")
-            try:
-                # Write bytes directly to buffer (Python 3).
-                sys.stdout.buffer.write(html)
-            except AttributeError:  # pragma: no cover
-                # Probably Python 2, which works with bytes by default.
-                sys.stdout.write(html)
+            html_bytes = html.encode(encoding, "xmlcharrefreplace")
+            # Write bytes directly to buffer (Python 3).
+            sys.stdout.buffer.write(html_bytes)
```

Comment:
Again, I agree that this is a reasonable update to old code, but it is out-of-scope for this PR.